feat: build and run from Docker Support

rajpratik71 opened the pull request:
Build and run the generator from docker image support added.

The main aim and motivation behind doing this is, because
of a no. of different jhipster blueprints are there,
which are compatible and works with the different version of jhipster
and also might be incompatible with other blueprints or versions of jhipster.

Signed-off-by: Pratik Raj <rajpratik71@gmail.com>

ctamisier commented:
Hi ! I have this:
![image](https://user-images.githubusercontent.com/766263/126904899-f5d08c91-5019-44ef-bdbe-6960a75b1e9f.png)

rajpratik71 committed:
feat: build and run from Docker Support

Build and run the generator from docker image support added.

The main aim and motivation behind doing this is, because
of a no. of different jhipster blueprints are there,
which are compatible and works with the different version of jhipster
and also might be incompatible with other blueprints or versions of jhipster.

Signed-off-by: Pratik Raj <rajpratik71@gmail.com>

rajpratik71 commented:
> Hi ! I have this:
> ![image](https://user-images.githubusercontent.com/766263/126904899-f5d08c91-5019-44ef-bdbe-6960a75b1e9f.png)

Hey , Sorry for mistake , forgetten to add "jhipster" user. Fixed and pushed now

ctamisier commented:
Thanks ! the image build works.

Running: `docker run -it --rm -v $PWD:/home/jhipster/app generator-jhipster-quasar-blueprint`

And then selecting all default values, I have some warnings:
![image](https://user-images.githubusercontent.com/766263/126906317-07550313-dd12-4e36-8aca-6a2cbfbb6319.png)

![image](https://user-images.githubusercontent.com/766263/126906331-c87d98e4-b745-4f6e-b187-68bd599dfc16.png)

![image](https://user-images.githubusercontent.com/766263/126906339-6a034f06-bd46-4ce1-aa0d-92293acc2127.png)

Should we just install these missing tools in the Dockerfile ?

rajpratik71 commented:
> Thanks ! the image build works.
> 
> Running: `docker run -it --rm -v $PWD:/home/jhipster/app generator-jhipster-quasar-blueprint`
> 
> And then selecting all default values, I have some warnings:
> ![image](https://user-images.githubusercontent.com/766263/126906317-07550313-dd12-4e36-8aca-6a2cbfbb6319.png)
> 
> ![image](https://user-images.githubusercontent.com/766263/126906331-c87d98e4-b745-4f6e-b187-68bd599dfc16.png)
> 
> ![image](https://user-images.githubusercontent.com/766263/126906339-6a034f06-bd46-4ce1-aa0d-92293acc2127.png)
> 
> Should we just install these missing tools in the Dockerfile ?

I am aware of these warnings and there is reason why i don't installed these tools in Container i.e. firstly these are not must needed tool or dependency for code generation or generator and futher these tools comes with a lot of dependencies e.g "keytool" comes with "JRE" which make container big in size , due to lots of dependencies , So to make container small and thin  , for easy pull and use , I ignored general tools which doesn't effect functionality.

ctamisier commented:
Yes, I get the point to keep a small image.
But in the main generator we have Java, git, ... : https://github.com/jhipster/generator-jhipster/blob/main/Dockerfile
And I think we can generate a new jhipster project with this docker image (without any warnings) ?
